task2/js/script.js

```diff
@@ -0,0 +1,125 @@
+function isArray(something) {
+  return something instanceof Array;
+}
+
+function isArray2(something) {
+  return Object.prototype.toString.call(something) === '[object Array]';
+}
+
+function range(from, to, step) {
+  var retArray = [];
+  var i;
+
+  if (from == undefined)
+    return retArray;
+
+  if (step == undefined)
+    step = 1;
+
+  if (to == undefined) {
+    to = from;
+    from = 0;
+  }
+
+  for (i = from; i < to; i += step)
+    retArray.push(i);
+
+  return retArray;
+}
+
+function compact(someArray) {
+  var retArray = someArray.map(function () {
+    return true;
+  });
+
+  return retArray;
+}
+
+function compact2(someArray) {
```

Comment:
names like someArray and retArray are weird
you could use result, source, compacted....